src/tls.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module use to handle TLS certificates for the NMS."""
+
+import logging
+from typing import Optional
+
+from charms.tls_certificates_interface.v4.tls_certificates import (
+    Certificate,
+    CertificateRequestAttributes,
+    PrivateKey,
+    TLSCertificatesRequiresV4,
+)
+from ops import CharmBase, Container
+
+logger = logging.getLogger(__name__)
+
+
+CERTIFICATE_COMMON_NAME = "nms.sdcore"
```

Comment:
I do not think we should hard code the common name. This seems to assume that the NMS will be deployed with the `nms` application name in a namespace named `sdcore`. We could find the name of the K8s service instead and set that.

Reply:
we are using the same pattern in all the NFs. Also notary uses just "Notary" so I thought it could be a name of our choice